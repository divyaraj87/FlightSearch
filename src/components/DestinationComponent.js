import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { fieldLoad } from '../actions';
import options from './SourceData';
import { Typeahead } from 'react-bootstrap-typeahead';
import 'react-bootstrap-typeahead/css/Typeahead.css';
import { DEST } from '../constants';

export class DestinationComponent extends Component {
    render() {
        const { dispatchSetSourceSelected, todos: { destSelected } } = this.props;
        return (
            <div class="DestContainer">To
             <Typeahead
                id="dest"
                onChange={(value) => dispatchSetSourceSelected(DEST, value)}
                options={options}
                placeholder="Select a city"
                selected={destSelected}
                />
            </div>
        );
    }
}

DestinationComponent.prototypes = {   
    dispatchSetSourceSelected: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    { dispatchSetSourceSelected: fieldLoad })(DestinationComponent);