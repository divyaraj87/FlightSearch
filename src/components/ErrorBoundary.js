import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';

export class ErrorBoundary extends Component {
    render() {
        const { todos: { error } } = this.props;
        return (
            <div class="error">
            {error}
            </div>
        );
    }
}

ErrorBoundary.prototypes = {
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    {})(ErrorBoundary);