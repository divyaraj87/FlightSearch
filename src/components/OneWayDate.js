import React, { useState } from "react";

import DatePicker from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";

export default function TableDatePicker() {
 const [startDate, setStartDate] = useState(new Date());
 const [endDate, setEndDate] = useState(new Date());

 return (
   <div className="dateContainer"><div className="start">Depart
     <DatePicker
       selected={startDate}
       selectsStart
       startDate={startDate}
       endDate={endDate}
       onChange={date => setStartDate(date)}
     /></div>
     <div className="end">End
     <DatePicker
       selected={endDate}
       selectsEnd
       startDate={startDate}
       endDate={endDate}
       minDate={startDate}
       onChange={date => setEndDate(date)}
     /></div>
   </div>
 );
}