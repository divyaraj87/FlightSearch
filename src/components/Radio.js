import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { fieldLoad } from '../actions';

export class Radio extends Component {
    render() {
        const { dispatchSetSourceSelected } = this.props;
        return (
            <div class="TripContainer">
                &nbsp;&nbsp;&nbsp;&nbsp;<input type="radio" id="trip" value="One Way" name="gender" onChange={(value) => dispatchSetSourceSelected('trip', this)}/> One Way &nbsp;&nbsp;&nbsp;&nbsp;
                <input type="radio" value="Return" name="gender" id="trip" onChange={(value) => dispatchSetSourceSelected('trip', this)}/> Return
            </div>
        );
    }
}

Radio.prototypes = {   
    dispatchSetSourceSelected: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    { dispatchSetSourceSelected: fieldLoad })(Radio);