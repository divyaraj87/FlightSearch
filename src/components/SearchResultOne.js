import React, { Component} from 'react';
import propTypes from 'prop-types';
import {connect } from 'react-redux';
import { fieldLoad } from '../actions';

export class SearchResultOne extends Component {
    render(){
        const {todos: { destSelected, sourceSelected }} = this.props;
        return (
            <div class="resultOneContainer">               
               <table>
                   <thead>
                       <tr>
                           <td colspan="3">{sourceSelected[0].label}</td>
                           <td colspan="1">TO</td>
                           <td colspan="3">{destSelected[0].label}</td>
                        </tr>
                       <tr>
                           <td>Flight No.</td>
                           <td>Flight Name</td>
                           <td>Departure Time</td>
                           <td>Arrival Time</td>
                           <td>Duration</td>
                           <td>No. of Stops</td>
                           <td>Price</td>
                        </tr>
                   </thead>
                   <tbody>
                       <tr>
                           <td>QF 171</td>
                           <td>Qantas Airways</td>
                           <td>13:15</td>
                           <td>23:10</td>
                           <td>7 hrs</td>
                           <td>1</td>
                           <td>1,400</td>
                       </tr>
                       <tr>
                           <td>AK 171</td>
                           <td>Emirates</td>
                           <td>3:30</td>
                           <td>23:10</td>
                           <td>22 hrs</td>
                           <td>2</td>
                           <td>2,300</td>
                       </tr>
                   </tbody>
                </table>            
            </div>
        );
    }
}

SearchResultOne.prototypes = {
    todos: propTypes.object,
    dispatchSearchFlight: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps, {
    dispatchSearchFlight: fieldLoad
})(SearchResultOne);