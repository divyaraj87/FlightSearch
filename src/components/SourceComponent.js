import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import { fieldLoad } from '../actions';
import options from './SourceData';
import { Typeahead } from 'react-bootstrap-typeahead';
import 'react-bootstrap-typeahead/css/Typeahead.css';
import { SOURCE } from '../constants';

export class SourceComponent extends Component {
    render() {
        const { dispatchSetSourceSelected, todos: { sourceSelected } } = this.props;
        return (
            <div class="SourceContainer">From
                <Typeahead
                    id="source"
                    onChange={(value) => dispatchSetSourceSelected(SOURCE, value)}
                    options={options}
                    placeholder="Select a city"
                    selected={sourceSelected}
                />
            </div>
        );
    }
}

SourceComponent.prototypes = {   
    dispatchSetSourceSelected: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps,
    { dispatchSetSourceSelected: fieldLoad })(SourceComponent);