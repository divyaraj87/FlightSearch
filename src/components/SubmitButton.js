import React, { Component} from 'react';
import propTypes from 'prop-types';
import {connect } from 'react-redux';
import { fieldLoad } from '../actions';

export class SubmitButton extends Component {
    render(){
        const {dispatchSearchFlight} = this.props;
        return (
            <div class="submitContainer">               
               <button class="submitBtn"            
                    onClick={() => dispatchSearchFlight('loadResponse', true)}>Search Flight</button>               
            </div>
        );
    }
}

SubmitButton.prototypes = {
    todos: propTypes.object,
    dispatchSearchFlight: propTypes.func
}

export const mapStateToProps = (state) => ({
    todos: state.todos
});

export default connect(mapStateToProps, {
    dispatchSearchFlight: fieldLoad
})(SubmitButton);