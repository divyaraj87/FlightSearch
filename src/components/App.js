import React, { Component } from 'react';
import propTypes from 'prop-types';
import { connect } from 'react-redux';
import  SourceComponent from './SourceComponent';
import  DestinationComponent from './DestinationComponent';
import Radio from './Radio';
import TableDatePicker from './OneWayDate';
import SubmitButton from './SubmitButton';
import ErrorBoundary from './ErrorBoundary';
import SearchResultOne from './SearchResultOne';
import SearchResultTwo from './SearchResultTwo';
import '../_style.scss';

export class App extends Component {
  render(){
      const { todos: { loadResponse, noReturn, error } } = this.props;
      return (
        <div className="container"> 
        <h1>Agent Application for Flight Booking</h1>
        <Radio /> 
        <div className="city">
           <SourceComponent />   
           <DestinationComponent /> 
          </div> 
         <div className="date">
            <TableDatePicker />
          </div> 
          <SubmitButton />
          {error != '' && <ErrorBoundary />}
          {loadResponse && <SearchResultOne /> }
          {loadResponse && noReturn && <SearchResultTwo /> }
      </div>
      );
  }
 
}

App.prototypes = {
  todos: propTypes.object
}

export const mapStateToProps = (state) => ({
  todos: state.todos
});

export default connect(mapStateToProps,
  { })(App);

