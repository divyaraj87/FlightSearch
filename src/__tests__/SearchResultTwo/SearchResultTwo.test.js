import React from 'react';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import  configureMockStore from 'redux-mock-store';
import { SearchResultTwo }  from '../../components/SearchResultTwo';

const dispatchSelectChangeSpy = sinon.spy();
const dispatchProgressChangeSpy = sinon.spy();
let mockProps = {
    bgcolor: '#000',
    completed: 10,
    width: 10,
    error: 'error',
    todos: {
      error: 'error'
    }
};
const mockStore = configureMockStore();
const store = mockStore({});
describe('SearchResultTwo', () => {
  let pageWrapper;

  describe('SearchResultTwo Page', () => {
    const originalProps = { ...mockProps };

    beforeEach(() => {
      pageWrapper = shallow(       
        <SearchResultTwo />       
      );
    });

    afterEach(() => {
      mockProps = { ...originalProps };
    });

    it('should render SearchResultTwo', () => {
      expect(pageWrapper).toMatchSnapshot();
    });
  });
});
