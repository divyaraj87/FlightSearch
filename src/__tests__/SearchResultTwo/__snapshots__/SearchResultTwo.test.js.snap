// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultTwo SearchResultTwo Page should render SearchResultTwo 1`] = `undefined`;
